Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.mm

```diff
@@ -459,10 +462,17 @@
     m_displayRefreshMonitorsToNotify = nullptr;
 }
 
-void RemoteLayerTreeDrawingArea::mainFrameContentSizeChanged(const IntSize& contentsSize)
+void RemoteLayerTreeDrawingArea::mainFrameContentSizeChanged(WebCore::FrameIdentifier frameID, const IntSize& contentsSize)
 {
-    // FIXME: Make this more aware of subframe processes where the root frame isn't always the main frame.
-    m_rootLayers[0].layer->setSize(contentsSize);
+    auto* webFrame = WebProcess::singleton().webFrame(frameID);
+    ASSERT(webFrame);
+    for (auto& rootLayerInfo : m_rootLayers) {
```

Comment:
Rather than the loop with confusing continue/return, using find() would be nicer.